moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -66,312 +61,279 @@ Eigen::Isometry3d createTestIsometry()
     return;
   }
-
-  // Robot state.
-  moveit::core::RobotState kinematic_state(robot_model);
-  const moveit::core::JointModelGroup* jmg = kinematic_state.getJointModelGroup(PANDA_TEST_GROUP);
-
-  // Provide our own random number generator to setToRandomPositions to get a deterministic sequence of joint
-  // configurations.
-  random_numbers::RandomNumberGenerator rng(0);
-
   KDL::Tree kdl_tree;
   if (!kdl_parser::treeFromUrdfModel(*robot_model->getURDF(), kdl_tree))
   {
     st.SkipWithError("Can't create KDL tree.");
     return;
   }
 
-  KDL::Chain kdl_chain;
-  if (!kdl_tree.getChain(jmg->getJointModels().front()->getParentLinkModel()->getName(),
-                         jmg->getLinkModelNames().back(), kdl_chain))
-  {
-    st.SkipWithError("Can't create KDL Chain.");
-    return;
-  }
+  KDL::TreeJntToJacSolver jacobian_solver(kdl_tree);
+  KDL::Jacobian jacobian(kdl_tree.getNrOfJoints());
+  KDL::JntArray kdl_q(kdl_tree.getNrOfJoints());
+  const std::string tip_link = jmg->getLinkModelNames().back();
 
-  KDL::ChainJntToJacSolver jacobian_solver(kdl_chain);
+  // Manually seeded RandomNumberGenerator for deterministic results
+  random_numbers::RandomNumberGenerator rng(0);
 
   for (auto _ : st)
   {
     // Time only the jacobian computation, not the forward kinematics.
     st.PauseTiming();
-    kinematic_state.setToRandomPositions(jmg, rng);
-    kinematic_state.updateLinkTransforms();
-    KDL::Jacobian jacobian(kdl_chain.getNrOfJoints());
-    KDL::JntArray kdl_q;
-    kdl_q.resize(kdl_chain.getNrOfJoints());
-    kinematic_state.copyJointGroupPositions(jmg, &kdl_q.data[0]);
+    state.setToRandomPositions(jmg, rng);
+    state.copyJointGroupPositions(jmg, &kdl_q.data[0]);
     st.ResumeTiming();
-    jacobian_solver.JntToJac(kdl_q, jacobian);
+    jacobian_solver.JntToJac(kdl_q, jacobian, tip_link);
   }
 }
 
-BENCHMARK(multiplyAffineTimesMatrix)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(multiplyMatrixTimesMatrix)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(multiplyIsometryTimesIsometry)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-
-BENCHMARK(inverseIsometry3d)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(inverseAffineIsometry)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(inverseAffine)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-BENCHMARK(inverseMatrix4d)->Arg(MATRIX_OPS_N_ITERATIONS)->Unit(benchmark::kMillisecond);
-
-BENCHMARK(robotStateConstruct)->RangeMultiplier(10)->Range(100, 10000)->Unit(benchmark::kMillisecond);
-BENCHMARK(robotStateCopy)->RangeMultiplier(10)->Range(100, 10000)->Unit(benchmark::kMillisecond);
-BENCHMARK(robotStateUpdate)->RangeMultiplier(10)->Range(10, 1000)->Unit(benchmark::kMillisecond);
-BENCHMARK(robotStateForwardKinematics)->RangeMultiplier(10)->Range(10, 1000)->Unit(benchmark::kMillisecond);
```

Comment:
See my other comment. I think it still makes sense to keep this one, but maybe rename to `robotStateGetGlobalLinkTransform`, since that's what it calls?